### PR TITLE
Add CSV export to new search

### PR DESCRIPTION
**Pull request by kmerz**
## Description
Add a button to the action list in the search which opens a modal. In the modal the user
can select the fields he wants to export and the stream he wants to filter the messages on.

**For 3.2.0 we will use the old backend to provide the CSV export.**

## Motivation and Context
CSV export is the only way to export messages via gui. So the user needs this feature really
badly.

This PR is based on: Refs #7214 

## Screenshots (if appropriate):
Graylog (46)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    }
+  }
+  return new URI(URLUtils.qualifyUrl('/notfound'));
+};
+
+const wrapOption = o => ({ label: o, value: o });
+const defaultFields = ['timestamp', 'source', 'message'];
+const defaultFieldOptions = defaultFields.map(wrapOption);
+
+const CSVExport = ({ closeModal, availableStreams, availableFields }: Props) => {
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx]: Sorting `availableStreams` and `availableFields` would really improve the usability.

File: graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+const SessionStore = StoreProvider.getStore('Session');
+
+type Option = {
+  label: String,
+  value: String,
+};
+
+type Props = {
+  availableStreams: Array<Option>,
+  availableFields: Array<Option>,
```
[the remark this answers is not in the document]
Comment (kmerz) [reply graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx]: :+1:

File: graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    ? 'Please right click the download link below and choose "Save Link As..." to download the CSV file.'
+    : 'Please click the download link below. Your browser may ask for your username and password to '
+    + 'download the CSV file.');
+
+  const Content = styled.div`
+    margin-left: 10px;
+    margin-right: 10px;
+  `;
+  return (
+    <Modal show>
```
[the remark this answers is not in the document]
Comment (kmerz) [reply graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx]: :+1:

File: graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmerz) [reply graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx]: :+1:

File: graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+const defaultFieldOptions = defaultFields.map(wrapOption);
+
+const CSVExport = ({ closeModal, availableStreams, availableFields }: Props) => {
+  const [selectedStream, setSelectedStream] = useState();
+  const [selectedFields, setSelectedFields] = useState(defaultFieldOptions);
+
+  const link = selectedFields.length > 0
+    ? (
+      /* eslint-disable-next-line react/jsx-no-target-blank */
+      <a href={getURLForExportAsCSV((selectedStream || {}).value, selectedFields.map(f => f.value))} target="_blank">
```
[the remark this answers is not in the document]
Comment (kmerz) [reply graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx]: :+1:

File: graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx (hunk 1 above)
Comment (kmerz) [reply graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx]: :+1:

**Pull request merged**